crates/hyperswitch_connectors/src/connectors/nexixpay/transformers.rs

```diff
@@ -90,50 +415,201 @@ impl TryFrom<&ConnectorAuthType> for NexixpayAuthType {
+#[serde(rename_all = "camelCase")]
+pub struct NexixpayPaymentsCaptureRequest {
+    amount: StringMinorUnit,
+    currency: Currency,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexixpayPaymentsCancelRequest {
+    description: Option<String>,
+    amount: StringMinorUnit,
+    currency: Currency,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexixpayOperationResponse {
+    operation_id: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct NexixpayRefundRequest {
+    pub amount: StringMinorUnit,
+    pub currency: Currency,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct RefundResponse {
+    operation_id: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexixpayErrorBody {
+    pub code: Option<String>,
+    pub description: Option<String>,
+}
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexixpayErrorResponse {
+    pub errors: Vec<NexixpayErrorBody>,
 }
 
-impl From<NexixpayPaymentStatus> for common_enums::AttemptStatus {
+impl From<NexixpayPaymentStatus> for AttemptStatus {
     fn from(item: NexixpayPaymentStatus) -> Self {
         match item {
-            NexixpayPaymentStatus::Succeeded => Self::Charged,
-            NexixpayPaymentStatus::Failed => Self::Failure,
-            NexixpayPaymentStatus::Processing => Self::Authorizing,
+            NexixpayPaymentStatus::Declined
+            | NexixpayPaymentStatus::DeniedByRisk
+            | NexixpayPaymentStatus::ThreedsFailed
+            | NexixpayPaymentStatus::Failed => Self::Failure,
+            NexixpayPaymentStatus::Authorized | NexixpayPaymentStatus::ThreedsValidated => {
+                Self::Authorized
+            }
+            NexixpayPaymentStatus::Executed => Self::Charged,
+            NexixpayPaymentStatus::Pending => Self::AuthenticationPending,
+            NexixpayPaymentStatus::Canceled | NexixpayPaymentStatus::Voided => Self::Voided,
+            NexixpayPaymentStatus::Refunded => Self::AutoRefunded,
         }
     }
 }
 
-//TODO: Fill the struct with respective fields
-#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
-pub struct NexixpayPaymentsResponse {
-    status: NexixpayPaymentStatus,
-    id: String,
+fn get_nexixpay_capture_type(
+    item: Option<CaptureMethod>,
+) -> CustomResult<Option<NexixpayCaptureType>, errors::ConnectorError> {
+    match item {
+        Some(CaptureMethod::Manual) => Ok(Some(NexixpayCaptureType::Explicit)),
+        Some(CaptureMethod::Automatic) => Ok(Some(NexixpayCaptureType::Implicit)),
+        Some(item) => Err(errors::ConnectorError::FlowNotSupported {
+            flow: item.to_string(),
+            connector: "Nexixpay".to_string(),
+        }
+        .into()),
+        None => Ok(None),
```

Comment:
If we pass None to the connector, what is the behaviour? automatic or manual
Map `None` to `Implicit` as this is the expected behaviour according to hyperswitch docs

Reply:
if none is passed it behaves `Implicit`